internal/resources/connections/resource_metrics_endpoint_scrape_job.go

```diff
@@ -226,3 +224,70 @@ func (r *resourceMetricsEndpointScrapeJob) Delete(ctx context.Context, req resou
 
 	resp.State.Set(ctx, nil)
 }
+
+type authBasicValidator struct{}
+
+func (v authBasicValidator) Description(ctx context.Context) string {
+	return "Validates that both username and password are provided for authentication basic"
+}
+
+func (v authBasicValidator) MarkdownDescription(ctx context.Context) string {
+	return "Validates that both username and password are provided for authentication basic"
+}
+
+func (v authBasicValidator) ValidateString(ctx context.Context, req validator.StringRequest, resp *validator.StringResponse) {
+	if req.ConfigValue.IsNull() || req.ConfigValue.IsUnknown() || req.ConfigValue.ValueString() != "basic" {
```

Comment:
It might be worth making `basic` and `bearer` const at this point given they are referenced in multiple places (e.g. https://github.com/grafana/terraform-provider-grafana/pull/2188/files#diff-e77468fffc9f8b719c4ff2070e8ff76fccc6baec5a4f96237f972f81794b54a5L93)

Reply:
Thanks, done!